tests/models/test_pyro.py

```diff
@@ -62,20 +47,58 @@ def __init__(self, in_features, out_features):
             .to_event(1)
         )
 
-    def forward(self, x, y):
-        sigma = pyro.sample("sigma", dist.Uniform(self.zero, self.ten))
+    def create_plates(self, x, y, ind_x):
+        return pyro.plate("data", size=self.n_obs, dim=-2, subsample=ind_x)
+
+    def list_obs_plate_vars(self):
+        """Create a dictionary with the name of observation/minibatch plate,
+        indexes of model args to provide to encoder,
+        variable names that belong to the observation plate
+        and the number of dimensions in non-plate axis of each variable"""
+
+        return {
+            "name": "obs_plate",
+            "in": [0],  # index for expression data
+            "sites": {},
+        }
+
+    @staticmethod
+    def _get_fn_args_from_batch(tensor_dict):
```

Comment:
what's the purpose of this in the pyro model as opposed to one level up (as before)?

Reply:
You mean what is the difference between `list_obs_plate_vars(self)` and `create_plates(self, x, y, ind_x)`?

The latter creates a pyro plate, the former records the name of the minibatch plate such that posterior sampling knows which is the minibatch plate (`"name": "obs_plate"`). Posterior sampling also needs to know which variables belong to that plate (keys of `"sites"` dictionary). The dimensions in `"sites"` dictionary and `"in"` are needed for the automatic autoencoding guide - but not strictly necessary for this PR.

WDYT?

Reply:
no I mean like why do you need this specific static method where the comment is?

Reply:
on a different note, should `list_obs_plate_vars(self)` be one level up in the `PyroBaseModuleClass` and just do like

```
def list_obs_plate_vars(self):
    pass
```

Reply:
@adamgayoso what do you think about this solution?
https://github.com/YosefLab/scvi-tools/pull/1059/commits/08dbe1f8c27073838343c79e70635ac3048f3f38